source/GameInterface/Services/MobileParties/Patches/CalculateBaseSpeedPatch.cs

```diff
@@ -0,0 +1,48 @@
+﻿using GameInterface.Services.MobileParties.Extensions;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.GameComponents;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.Core;
+
+namespace GameInterface.Services.MobileParties.Patches;
+
+[HarmonyPatch(typeof(DefaultPartySpeedCalculatingModel))]
+internal class CalculateBaseSpeedPatch 
+{
+    [HarmonyPatch(nameof(DefaultPartySpeedCalculatingModel.CalculateBaseSpeed))]
+    [HarmonyPostfix]
+    private static void CalculateBaseSpeed(ref MobileParty mobileParty, ref ExplainedNumber __result)
+    {
+        if(ModInformation.IsServer && mobileParty.IsPartyControlled() == false)
+        {
+            float playerMapMovementSpeedBonusMultiplier = Campaign.Current.Models.DifficultyModel.GetPlayerMapMovementSpeedBonusMultiplier();
+            if (playerMapMovementSpeedBonusMultiplier > 0f)
+            {
+                __result.AddFactor(playerMapMovementSpeedBonusMultiplier, GameTexts.FindText("str_game_difficulty"));
+            }
+        }
+    }
+}
+
+[HarmonyPatchCategory(GameInterface.HARMONY_STATIC_FIXES_CATEGORY)]
```

Comment:
```suggestion
/// <summary>
/// Fixes issue with DefaultPartySpeedCalculatingModel._culture statically calls GameTexts.FindText
/// and when harmony patches, it calls the static constructor for DefaultPartySpeedCalculatingModel
/// and results in a null reference exception because _gameTextManager has not been initialized
/// This patch initializes _gameTextManager if it is null
/// </summary>
[HarmonyPatchCategory(GameInterface.HARMONY_STATIC_FIXES_CATEGORY)]
```